src/cart/crt_swim.c

```diff
@@ -1534,6 +1536,10 @@ void crt_swim_rank_del_all(struct crt_grp_priv *grp_priv)
 			SWIM_STATUS_CHARS[cst->cst_state.sms_status],
 			cst->cst_state.sms_incarnation);
 		D_FREE(cst);
+
+		crt_swim_csm_unlock(csm);
```

Comment:
so you're unlocking then deleting then locking back ?

Reply:
@soumagne, that's right. We lock `csm` in order to delete a rank from `crt_swim`-level tracking structures. Then, we want to delete states of this rank in the `swim` level, which requires locking `csm_ctx` (inside `swim_member_del`). Due to lock order considerations, we have to unlock `csm` first.

I think ideally we should remove locking from the `swim` module and rely solely on locking in the `crt_swim` module. Then the lock order considerations here would disappear, and the deletion would become one atomic change. Probably too much for the current problem and patch.